### ISSUE TITLE
Workarounds for fixed WindowsPipe bugs no longer needed?

### ISSUE THREAD

**Issue by flip111** [symfony/symfony#14533]
1) Fixed bug
https://github.com/symfony/symfony/blob/dbbe1701680bfabb64568f2731e7098ffe2ebace/src/Symfony/Component/Process/Pipes/WindowsPipes.php#L90

https://github.com/symfony/symfony/blob/dbbe1701680bfabb64568f2731e7098ffe2ebace/src/Symfony/Component/Process/Pipes/WindowsPipes.php#L46-L49

http://git.php.net/?p=php-src.git;a=commit;h=0c982798e01e5a48765aa5352f96066e77e36efd

2) Never was a bug, bad bug report
https://github.com/symfony/symfony/blob/dbbe1701680bfabb64568f2731e7098ffe2ebace/src/Symfony/Component/Process/Pipes/WindowsPipes.php#L91

https://bugs.php.net/bug.php?id=65650 (last comments)

**Comment by javiereguiluz** [comment]
@romainneutron @nicolas-grekas is it worth it to remove these Windows workarounds?

**Comment by nicolas-grekas** [comment]
But our lowest PHP version in 5.5.9, and the fix landed in 5.5.18, so we can't remove any workaround I guess. Otherwise @flip111, don't hesitate to open a PR.

**Comment by flip111** [comment]
You can remove the workaround by placing conditionals on the php version. Which would not hurt performance (because of optimization in opcache). But the code will become ugly because the responsibility of handling windows pipes is not only contained within the WindowsPipe class but also needs special conditionals in the Process class https://github.com/symfony/symfony/blob/d55114ef3413d20031917a0dc9d8e1672f1aff8b/src/Symfony/Component/Process/Process.php#L264-L274

**Comment by stof** [comment]
The last comment you mentionned to say it was not a PHP bug looks suspicious to me. Having to wait until the process is ended to read the output would break the features allowing to read the input incrementally and to process the output while the process is running (to forward it to the console output for instance).

**Comment by flip111** [comment]
I assume you are referring to

> This is not a bug in PHP, but rather a timing issue in the supplied
test script. You have to wait until the other process has finished,
before you can read all its stdout and stderr. See Bill's code (the
while ($status['running']) loop) for a possible solution.

> Furthermore it is safer to call proc_close only after all pipes
haven been closed.

I agree it's not really clear. Especially with a nuance on `all` in `before you can read all its stdout and stderr`, which would indicate that you can already start reading it partially earlier ...

I would wait with any PR until someone has an actual performance issue and/or minimum php version for symfony has been bumped to 5.5.18 or higher. I'm just saying this because i think it will not be easy understand the complexity of what is going on. In case someone can fix it easily, it would be nice .. Same goes for nr 1

**Comment by javiereguiluz** [comment]
I'm closing this for the reasons given by @nicolas-grekas. Anyway, @flip111 thanks for opening this issue and for investigating new ways of improving Symfony.
